src/binary-reader.cc

```diff
@@ -2186,8 +2186,7 @@ Result BinaryReader::ReadElemSection(Offset section_size) {
     }
 
     // For backwards compat we support not declaring the element kind.
-    bool legacy = !(flags & SegPassive) && !(flags & SegExplicitIndex);
-    if (!legacy) {
+    if (flags & (SegPassive | SegExplicitIndex)) {
```

Comment:
This is a no-op right?

Reply:
Yep, no difference. I found this a bit clearer to compare to the table in the doc, but it's still not super clear.